x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs_logic.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { DEFAULT_META } from '../../../shared/constants';
+import { flashAPIErrors, setErrorMessage } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { updateMetaPageIndex } from '../../../shared/table_pagination';
+import { EngineLogic } from '../engine';
+
+import { POLLING_DURATION, POLLING_ERROR_MESSAGE } from './constants';
+import { ApiLogsData, ApiLog } from './types';
+import { getDateString } from './utils';
+
+interface ApiLogsValues {
+  dataLoading: boolean;
+  apiLogs: ApiLog[];
+  meta: ApiLogsData['meta'];
+  hasNewData: boolean;
+  polledData: ApiLogsData;
+  intervalId: number | null;
+}
+
+interface ApiLogsActions {
+  fetchApiLogs(options?: { isPoll: boolean }): { isPoll: boolean };
+  pollForApiLogs(): void;
+  onPollStart(intervalId: number): { intervalId: number };
+  onPollInterval(data: ApiLogsData): ApiLogsData;
+  storePolledData(data: ApiLogsData): ApiLogsData;
+  updateView(data: ApiLogsData): ApiLogsData;
+  onUserRefresh(): void;
+  onPaginate(newPageIndex: number): { newPageIndex: number };
```

Comment:
I played around with the names a bit more from our Slack convo - let me know what you think! I think it is a bit clearer in terms of imperative vs declarative naming (tying user actions/flows to fn names) but definitely open to suggestions, naming things is hard 😬

Reply:
I'm good with w/e - I've been using `onPaginate` in other places w/ table pagination (overview, curations, etc.) so we should probably change them all at once. Definitely open to a separate follow-up PR for this